src/main/java/com/minecolonies/core/colony/buildings/modules/TavernBuildingModule.java

```diff
@@ -185,10 +186,15 @@ private void spawnVisitorInternal()
     /**
      * Spawns a visitor citizen that can be recruited.
      */
+    @Nullable
     public IVisitorData spawnVisitor()
     {
         final int recruitLevel = building.getColony().getWorld().random.nextInt(10 * building.getBuildingLevel()) + 15;
         final RecruitCost cost = RecruitmentItemsListener.getRandomRecruitCost(building.getColony().getWorld().getRandom(), recruitLevel);
+        if (cost == null)
```

Comment:
why would the recuitcost be null?

Reply:
At the off-chance there are no data files with recruit costs or the filters return nothing, we don't want NPE's happening

Reply:
We need to log the cause of this then, recruit cost should never be null

Reply:
Added a log statement in the listener class